server/mediaserver_test.go

```diff
@@ -278,48 +310,63 @@ func TestCreateRTMPStreamHandlerWebhook(t *testing.T) {
 	defer ts.Close()
 	AuthWebhookURL = ts.URL
 	sid = createSid(u)
-	if sid == "" {
-		t.Error("On empty response with 200 code should pass")
-	}
-	ts2 := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte(`{"manifestID":""}`))
-	}))
-	defer ts.Close()
-	AuthWebhookURL = ts2.URL
+	assert.NotNil(sid, "On empty response with 200 code should pass")
+
+	// local helper to reduce boilerplate
+	makeServer := func(resp string) *httptest.Server {
+		t := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Write([]byte(resp))
+		}))
+		AuthWebhookURL = t.URL
+		return t
+	}
+	defer func() { AuthWebhookURL = "" }()
+	BroadcastJobVideoProfiles = []ffmpeg.VideoProfile{ffmpeg.P360p30fps16x9}
+
+	// empty manifestID
+	ts2 := makeServer(`{"manifestID":""}`)
+	defer ts2.Close()
 	sid = createSid(u)
-	if sid != "" {
-		t.Error("Should not pass in returned manifest id is empty")
-	}
-	ts3 := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte(`{manifestID:"XX"}`))
-	}))
+	assert.Nil(sid, "Should not pass if returned manifest id is empty")
+
+	// invalid json
+	ts3 := makeServer(`{manifestID:"XX"}`)
 	defer ts3.Close()
-	AuthWebhookURL = ts3.URL
 	sid = createSid(u)
-	if sid != "" {
-		t.Error("Should not pass if returned json is invalid")
-	}
-	ts4 := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte(`{"manifestID":"xy"}`))
-	}))
+	assert.Nil(sid, "Should not pass if returned json is invalid")
+
+	// set manifestID
+	ts4 := makeServer(`{"manifestID":"xy"}`)
 	defer ts4.Close()
-	AuthWebhookURL = ts4.URL
-	sid = createSid(u)
-	mid := parseManifestID(sid)
-	if mid != "xy" {
-		t.Error("Should set manifest id to one provided by webhook")
-	}
-	ts5 := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte(`{"manifestID":"xyz", "streamKey":"zyx"}`))
-	}))
+	params := createSid(u).(*streamParameters)
+	mid := params.mid
+	assert.Equal(core.ManifestID("xy"), params.mid, "Should set manifest id to one provided by webhook")
+
+	// ensure the presets match defaults
+	assert.Len(params.profiles, 1)
+	assert.Equal(params.profiles, BroadcastJobVideoProfiles, "Default presets did not match")
+
+	// set manifestID + streamKey
+	ts5 := makeServer(`{"manifestID":"xyz", "streamKey":"zyx"}`)
 	defer ts5.Close()
-	AuthWebhookURL = ts5.URL
-	sid = createSid(u)
-	mid = parseManifestID(sid)
-	if mid != "xyz" || sid != "xyz/zyx" {
-		t.Error("Should set manifest / streamkey  to one provided by webhook")
-	}
-	AuthWebhookURL = ""
+	params = createSid(u).(*streamParameters)
+	mid = params.mid
+	assert.Equal(core.ManifestID("xyz"), mid, "Should set manifest to one provided by webhook")
+	assert.Equal("xyz/zyx", params.StreamID(), "Should set streamkey to one provided by webhook")
```

Comment:
Do we want to check `params.rtmpKey` as well?

Reply:
Sure. This was just a conversion of an older set of tests, rather than adding new ones, but added a new one here.